agents-common/src/main/java/org/apache/ranger/plugin/util/SearchFilter.java

```diff
@@ -98,6 +100,8 @@ public class SearchFilter {
 	private Map<String, String> params;
 	private int                 startIndex;
 	private int                 maxRows    = Integer.MAX_VALUE;
+	private int                 beginIndex = -1;
+	private int                 offsetIndex = -1;
```

Comment:
Since you've added new fields to the SearchFilter class, don't forget to modify the copy constructor (public SearchFilter(SearchFilter other)) accordingly to ensure the new attributes are properly copied.